lab_3/main.py

```diff
@@ -110,24 +188,66 @@ def new_language(self, encoded_text: tuple, language_name: str) -> int:
         :param language_name: a language
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(encoded_text, tuple) or \
+                None in encoded_text or not isinstance(language_name, str):
```

Comment:
`None in encoded_text` too specific. You should check that passed argument is _appropriate_.